src/neurostatslib/basis.py

```diff
@@ -533,6 +537,218 @@ def _check_n_basis_min(self) -> None:
+            The basis function evaluated at the samples (Time points x number of basis)
+
+        Raises
+        ------
+        AssertionError
+            If the sample points are not within the B-spline knots range unless `outer_ok=True`.
+
+        Notes
+        -----
+        This method evaluates the B-spline basis functions at the given sample points.
+        It requires the knots to be defined through the `_generate_knots` method.
+
+        The evaluation is performed by looping over each element and using `splev`
+        from SciPy to compute the basis values.
+        """
+        super()._check_samples_non_empty(sample_pts)
+
+        # add knots
+        self._generate_knots(sample_pts, 0.0, 1.0)
+
+        basis_eval = bspline(
+            sample_pts, self.knot_locs, order=self.order, der=0, outer_ok=False
+        )
+
+        return basis_eval
+
+    def _check_n_basis_min(self) -> None:
+        """Check that the user required enough basis elements.
+
+        Check that BSplineBasis has at least as many basis as the order of the spline.
+
+        Raises
+        ------
+        ValueError
+            If an insufficient number of basis element is requested for the basis type
+        """
+        if self.n_basis_funcs < self.order:
+            raise ValueError(
+                f"{self.__class__.__name__} `order` parameter cannot be larger "
+                "than `n_basis_funcs` parameter."
+            )
+
+
+class CyclicBSplineBasis(SplineBasis):
+    """
+    B-spline 1-dimensional basis functions for cyclic splines.
+
+    Parameters
+    ----------
+    n_basis_funcs :
+        Number of basis functions.
+    order :
+        Order of the splines used in basis functions. Must lie within [1, n_basis_funcs].
+        The B-splines have (order-2) continuous derivatives at each interior knot.
+        The higher this number, the smoother the basis representation will be.
+
+    Attributes
+    ----------
+    n_basis_funcs : int
+        Number of basis functions.
+    order : int
+        Order of the splines used in basis functions.
+
+    Methods
+    -------
+    _evaluate(sample_pts, der)
+        Evaluate the B-spline basis functions with given sample points.
+    """
+
+    def __init__(self, n_basis_funcs: int, order: int = 2):
+        super().__init__(n_basis_funcs, order=order)
+        if self.order < 2:
+            raise ValueError(
+                f"Order >= 2 required for cyclic B-spline, "
+                f"order {self.order} specified instead!"
+            )
+
+    def _evaluate(self, sample_pts: NDArray) -> NDArray:
+        """
+        Evaluate the B-spline basis functions with given sample points.
+
+        Parameters
+        ----------
+        sample_pts :
+            The sample points at which the B-spline is evaluated. Must be a tuple of length 1.
+
+        Returns
+        -------
+        NDArray
+            The basis function evaluated at the samples (Time points x number of basis)
```

Comment:
```suggestion
            The basis function evaluated at the samples, shape (n_samples, n_basis_funcs)
```
to be consistent with other basis objects